cli/src/main/java/com/liferay/blade/cli/BladeCLI.java

```diff
@@ -385,17 +394,121 @@ public void run(String[] args) throws Exception {
 
 						Object commandArgs = objects.get(0);
 
-						_command = command;
+						Console console = System.console();
+
+						if (console != null) {
+							while (parameterException != null) {
+								String parameterMessage = parameterException.getMessage();
+
+								List<String> argsCollection = new ArrayList<>(Arrays.asList(args));
+
+								if (parameterMessage.contains("The following options are required: ") ||
+									parameterMessage.contains("The following option is required: ")) {
+
+									parameterMessage = parameterMessage.replace(
+										"The following options are required: ", "");
+									parameterMessage = parameterMessage.replace(
+										"The following option is required: ", "");
+
+									String[] missingParameters = parameterMessage.split(", ");
+
+									for (String missingParameter : missingParameters) {
+										if (missingParameter.contains(" | ")) {
+											missingParameter = missingParameter.split(" | ")[0];
+										}
+
+										if (missingParameter.startsWith("[")) {
+											missingParameter = missingParameter.substring(1);
+										}
+
+										if (missingParameter.endsWith("]")) {
+											missingParameter = missingParameter.substring(
+												0, missingParameter.length() - 1);
+										}
+
+										_promptForMissingParameter(commandArgs, argsCollection, missingParameter);
+									}
+
+									args = argsCollection.toArray(new String[0]);
+
+									args = Extensions.sortArgs(_commands, args);
+
+									try {
+										parameterException = null;
+
+										_jCommander = _buildJCommanderWithCommandMap(_commands);
+
+										_jCommander.parse(args);
+									}
+									catch (ParameterException pe) {
+										parameterException = pe;
+									}
+
+									jCommands = _jCommander.getCommands();
+
+									jCommander = jCommands.get(command);
+
+									if (jCommander == null) {
+										printUsage();
+
+										break;
+									}
+
+									objects = jCommander.getObjects();
+
+									commandArgs = objects.get(0);
+								}
+								else if (parameterMessage.contains("Main parameters are required")) {
+									_promptForMissingMainParameter(commandArgs, argsCollection);
+
+									args = argsCollection.toArray(new String[0]);
```

Comment:
The following lines 464 - 491 look very similar to 432 - 459 can they be combined?